.github/workflows/main_ci.yml

```diff
@@ -0,0 +1,94 @@
+name: Python Testing
+
+on:
+  push:
+    branches:
+      - pydarshan-devel
+  pull_request:
+    branches:
+      - pydarshan-devel
```

Comment:
I can do it if you want, but do you mind if we change this branch name to `main` here and above.

If I see the CI run for this PR that would be useful in terms of the number of tests that run vs. current `pydarshan-devel` locally, etc.

Reply:
I just pushed a commit for this.

It may already be the case, but I'm assuming there's a way to restrict these tests so they only run on changes to darshan-util/pydarshan? That should help restrict running tests when nothing really has changed that would impact pydarshan.

Reply:
Yeah, probably any changes to `darshan-util`, as well as `include` which has the log format headers.